libs/types/src/cdf/election-results-reporting/convert.ts

```diff
@@ -378,10 +399,25 @@ function validateCandidateIds(
         'No ContestSelections defined for CandidateContest'
       ) as ResultsReporting.CandidateSelection[];
       for (const selection of contestSelections) {
-        if (!selection.IsWriteIn && !validCandidateIds.has(selection['@id'])) {
+        if (selection.IsWriteIn) {
+          continue;
+        }
+
+        const candidateId = trimVxIdPrefix(
+          assertDefined(
+            selection.CandidateIds,
+            `No candidate ID on selection: ${JSON.stringify(
+              selection,
+              null,
+              2
+            )}`
+          )[0]
```

Comment:
`CandidateIds` is not required by the schema but I can't think of any other way to relate `CandidateSelection` to `Candidate` except packing that info in `CandidateSelection['@id']` and then parsing the ID string. `CandidateIds` feels like it was designed for this purpose. If it's not de-factor required by industry convention, I think we should at least require it in our own products.

Reply:
I think this is the way the format is intended to be used. My guess is that maybe `CandidateIds` is not required because a write-in selection might not align to any candidate.

I've noticed in CDF docs that UML fields that are linked by ID usually don't have the `ID` suffix in the docs. I think that's a pattern.